WalletWasabi.Gui/UiConfig.cs

```diff
@@ -17,129 +17,145 @@
+        {
+            AssertFilePathSet();
+
+            WindowState = Avalonia.Controls.WindowState.Maximized;
+            Height = 530;
+            Width = 1100;
+            FeeTarget = 2;
+            FeeDisplayFormat = 0;
+            Autocopy = true;
+            LurkingWifeMode = false;
+            LockScreenActive = false;
+            LockScreenPinHash = "";
+
+            if (!File.Exists(FilePath))
+            {
+                Logging.Logger.LogInfo<Config>($"{nameof(Config)} file did not exist. Created at path: `{FilePath}`.");
+            }
+            else
+            {
+                await LoadFileAsync();
+            }
+
+            await ToFileAsync();
+        }
+
+        public async Task LoadFileAsync()
+        {
+            string jsonString = await File.ReadAllTextAsync(FilePath, Encoding.UTF8);
+            var config = JsonConvert.DeserializeObject<UiConfig>(jsonString);
+
+            WindowState = config.WindowState ?? WindowState;
+            Height = config.Height ?? Height;
+            Width = config.Width ?? Width;
+            FeeTarget = config.FeeTarget ?? FeeTarget;
+            FeeDisplayFormat = config.FeeDisplayFormat ?? FeeDisplayFormat;
+            Autocopy = config.Autocopy ?? Autocopy;
+            LurkingWifeMode = config.LurkingWifeMode ?? LurkingWifeMode;
+            LockScreenActive = config.LockScreenActive ?? LockScreenActive;
+            LockScreenPinHash = config.LockScreenPinHash ?? LockScreenPinHash;
+        }
+
+        /// <inheritdoc />
+        public async Task<bool> CheckFileChangeAsync()
+        {
+            AssertFilePathSet();
+
+            if (!File.Exists(FilePath))
+            {
+                throw new FileNotFoundException($"{nameof(Config)} file did not exist at path: `{FilePath}`.");
+            }
+
+            string jsonString = await File.ReadAllTextAsync(FilePath, Encoding.UTF8);
 			var newConfig = JsonConvert.DeserializeObject<JObject>(jsonString);
 			var currentConfig = JObject.FromObject(this);
 
 			return !JToken.DeepEquals(newConfig, currentConfig);
-		}
-
-		/// <inheritdoc />
-		public void SetFilePath(string path)
-		{
-			FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(path), path, trim: true);
-		}
-
-		/// <inheritdoc />
-		public void AssertFilePathSet()
-		{
-			if (FilePath is null)
-			{
-				throw new NotSupportedException($"{nameof(FilePath)} is not set. Use {nameof(SetFilePath)} to set it.");
-			}
-		}
-	}
+        }
+
+        /// <inheritdoc />
+        public void SetFilePath(string path)
+        {
+            FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(path), path, trim: true);
+        }
+
+        /// <inheritdoc />
+        public void AssertFilePathSet()
+        {
+            if (FilePath is null)
+            {
+                throw new NotSupportedException($"{nameof(FilePath)} is not set. Use {nameof(SetFilePath)} to set it.");
+            }
+        }
+    }
```

Comment:
Wooh, what's going on here?

Reply:
VSCode uses 4 spaces rather than tabs.. will replace those after i've addressed the non-formatting related issues.

Reply:
Hehe, Adam is all about tabs, not spaces ;)